Version 1.0.0

ihorserba opened the pull request.

vocoded commented on sdks/js/package.json:
```diff
@@ -1,7 +1,7 @@
 {
   "name": "zello-channels-js-sdk",
   "copyright": "Copyright © 2021 Zello Inc.",
```
let's update the copyright as well

ihorserba replied:
Updated

vocoded commented on sdks/js/package.json:
```diff
@@ -36,7 +36,7 @@
     "rename-output-webpack-plugin": "^1.0.1",
     "sass-loader": "^7.0.1",
-    "scriptjs": "^2.5.8",
+    "scriptjs": "^2.5.9",
```
are there any other recommended dependency updates we should catch here?

ihorserba replied:
Not easily, but I did some progress: https://github.com/zelloptt/zello-channel-api/pull/187

vocoded commented on sdks/README.md:
```diff
@@ -297,6 +297,7 @@ In iOS, the events interface is `ZCCSessionDelegate`.
 |`Session.incoming_voice_will_start`|This event is fired when an incoming voice stream is about to start. `IncomingMessage` object is returned
 |`Session.incoming_voice_did_start`|Incoming voice message did start (first packet received)
 |`Session.incoming_voice_did_stop`|An incoming voice stream has finished playing.
+|`Session.outgoing_voice_did_stop`|An outgoing voice stream has been stopped by the server
```
let's add the other one (fail_start)

ihorserba replied:
Added

vocoded reviewed (commented):
Final minor requests, then we're good

adambailey- commented on sdks/js/src/classes/player.js:
```diff
@@ -13,5 +13,5 @@
  */
-const PCMPlayer = require('./../vendor/pcm-player/pcm-player');
+const PCMPlayer = require('./../vendor/pcm-player/pcm-player.min');
 module.exports = PCMPlayer;
```
add newline

ihorserba replied:
Added

ihorserba commented:
Once https://github.com/zelloptt/opus-recorder/pull/7 is re-approved (there were some changes since the last approve) - we have to update the `opus-recorder` submodule

ihorserba commented:
Last fixes for recorder: https://github.com/zelloptt/opus-recorder/pull/9

adambailey- reviewed (commented):
Is [80e753f52562eb454910.worker.js](https://github.com/zelloptt/zello-channel-api/pull/184/files#diff-5d69eefe41f40f06ba17bfbc9fd72ac15579dfeed62941db1be031e4a6ec4f31) a desired file we actually want to commit? It seems to me like these files are maybe incidental from our build process, and that we don't actually want them?

ihorserba commented:
Another small documentation fix
https://github.com/zelloptt/opus-to-pcm/pull/2

adambailey- commented:
> It's requested as blob before sending and before receiving audio data.

What's the significance behind the name of this?

vocoded commented:
@adambailey- Nothing we are setting, this is output from emscripten

ihorserba commented:
> > It's requested as blob before sending and before receiving audio data.
> 
> What's the significance behind the name of this?

I'm not sure.
The filename is generated by webpack.

ihorserba commented:
Back-merged master
Merged the [auth token version support](https://github.com/zelloptt/zello-wss/pull/120)
Rebuilt SDK

vocoded commented on sdks/README.md:
```diff
@@ -295,8 +295,10 @@ In iOS, the events interface is `ZCCSessionDelegate`.
 |`Session.incoming_voice_will_start`|This event is fired when an incoming voice stream is about to start. `IncomingMessage` object is returned
-|`Session.incoming_voice_did_start`|Incoming voice message did start (first packet received)
+|`Session.incoming_voice_did_start`|Incoming voice message did start (first packet received and the playback has been initialized correctly)
+|`Session.incoming_voice_fail_start`|This event is called if the audio data was received, but the playback has been failed to start
```
```suggestion
|`Session.incoming_voice_fail_start`|This event is called if the audio data was received, but playback failed to start
```

ihorserba replied:
Changed

vocoded commented on sdks/js/src/classes/sdk.js:
```diff
@@ -84,6 +86,14 @@ ZCC.Sdk.init({
       widget: false
     }, options);
 
+    let scriptsNames = [
```
minor, reads better as `scriptNames` (no need for both to be plural)

ihorserba replied:
Changed

vocoded reviewed (approved):
minor. are you still waiting on anything to merge?

ihorserba commented:
Back-merged master, changed according to the latest minor comments; rebuilt SDK

ihorserba commented:
> minor. are you still waiting on anything to merge?

I think, now we're good to merge

ihorserba commented:
> minor. are you still waiting on anything to merge?

Oh, I remember the comment [here](https://github.com/zelloptt/zello-channel-api/pull/193) - Maybe add a brief description about possible `tokenVersion` values? Or it's supposed not to be used by SDK users yet?

vocoded commented:
It can be used, the only supported value is `v1` to indicate the current token format. if omitted, this will be the assumed version.
